src/core/thread/mle.cpp

```diff
@@ -1744,6 +1727,30 @@ bool Mle::PrepareAnnounceState(void)
     return shouldAnnounce;
 }
 
+otError Mle::GetActiveDatasetChannelMask(Mac::ChannelMask &aChannelMask)
```

Comment:
There seems to be another `GetActiveDatasetChannelMask()` in [src/core/thread/announce_sender.cpp](https://github.com/openthread/openthread/blob/c06179938c8d5deb59822f28482304e2589baf69/src/core/thread/announce_sender.cpp#L125). Can we combine these into a single implementation?

Reply:
How about adding this method to `ActiveDataset` or `DatasetManager` class?

Reply:
Great suggestion. I have added a method to `DatasetManager ` class.